ui/src/components/admin/sermonNotes/SermonNotesEditorModal.js

```diff
@@ -0,0 +1,384 @@
+  };
+
+  const formatDate = (date) => {
+    const parts = date.split('-');
+    const day = parts[2];
+    const month = parts[1];
+    const year = parts[0];
+    return `${day}${month}${year}`;
+  };
+
+  const fetchSermonNotes = async () => {
+    try {
+      const { data } = await axios.get('/api/sermon-notes-parent/get');
+      return data;
+    } catch (err) {
+      console.log(err);
+      toast({
+        description:
+          'There was an issue with the request, please talk to a t3ch support',
+        status: 'error',
+        duration: 8000,
+        isClosable: true,
+      });
+    }
+  };
+
+  const onSubmitSermonNotes = async (e) => {
+    try {
+      if (actionOnEditor === 'edit') {
+        const { status } = await axios.put('/api/sermon-notes-parent/update', {
+          sermonId: editSermonNotesData.sermonId,
+          title,
+          subtitle,
+          speaker,
+          sermonSeries,
+          date,
+          imageLink,
+          originalContent,
+          sermonLink,
+          serviceType,
+          passage,
+        });
+        if (status === 200) {
+          toast({
+            title: 'Announcement Updated',
+            description: 'Your announcement has been updated.',
+            status: 'success',
+            duration: 5000,
+            isClosable: true,
+          });
+        }
+        setIsEditorOpen(false);
+      } else {
+        // Create Unique Sermon ID
+        const formattedData = formatDate(date);
+        const sermonId = `${sermonIdMap[serviceType]}-${formattedData}-${
+          numberOfSermons + 1
+        }`;
+        setSermonId(sermonId);
+        setValue('sermonId', sermonId);
+
+        const { status } = await axios.post('/api/sermon-notes-parent/create', {
+          sermonId,
+          title,
+          subtitle,
+          speaker,
+          sermonSeries,
+          date,
+          imageLink,
+          originalContent,
+          sermonLink,
+          serviceType,
+          passage,
+          isPublished: true,
+        });
+        if (status === 200) {
+          toast({
+            title: 'Announcement Created',
+            description: 'Your announcement has been created.',
+            status: 'success',
+            duration: 5000,
+            isClosable: true,
+          });
+        }
+        setIsEditorOpen(false);
+      }
+    } catch (err) {
+      console.log('Error');
+      toast({
+        title: 'Error',
+        description: 'An error occurred. Please try again later.',
+        status: 'error',
+        duration: 5000,
+        isClosable: true,
+      });
+    }
+  };
+
+  const getData = async () => {
+    const result = await fetchSermonNotes();
+    const numberOfSermonNotes = result.filter((sermonNote) => {
+      return sermonNote.date === date;
+    });
+    if (numberOfSermonNotes.length > 0) {
+      setNumberOfSermons(numberOfSermonNotes.length);
+    } else {
+      setNumberOfSermons(0);
+    }
+  };
```

Comment:
you can add comments explaining here that the numberOfSermonNotes you are referring too is for the same dates, out of context it can be confusing